domain/modelagent/state/model/state_test.go

```diff
@@ -1523,3 +1583,82 @@ func (s *modelStateSuite) TestUpdateLatestAgentVersionLessThanCurrentTarget(c *t
 	err := st.UpdateLatestAgentVersion(c.Context(), semversion.MustParse("4.1.0"))
 	c.Assert(err, tc.ErrorIs, modelagenterrors.LatestVersionDowngradeNotSupported)
 }
+
+func (s *modelStateSuite) TestGetAllMachineBases(c *tc.C) {
+	st := NewState(s.TxnRunnerFactory())
+
+	machineUUID1 := s.addMachineWithBase(c, corebase.MakeDefaultBase(corebase.UbuntuOS, "24.04"))
+	machineBases, err := st.GetAllMachineBases(c.Context())
+	c.Assert(err, tc.ErrorIsNil)
+	c.Assert(len(machineBases), tc.Equals, 1)
+
+	machineUUID2 := s.addMachineWithBase(c, corebase.MakeDefaultBase(corebase.UbuntuOS, "25.04"))
+	machineBases, err = st.GetAllMachineBases(c.Context())
+	c.Assert(err, tc.ErrorIsNil)
+	c.Assert(len(machineBases), tc.Equals, 2)
+
+	c.Assert(machineBases[machineUUID1.String()].Channel.String(), tc.Equals, "24.04/stable")
+	c.Assert(machineBases[machineUUID1.String()].OS, tc.Equals, "ubuntu")
+	c.Assert(machineBases[machineUUID2.String()].Channel.String(), tc.Equals, "25.04/stable")
+	c.Assert(machineBases[machineUUID2.String()].OS, tc.Equals, "ubuntu")
+}
+
+func (s *modelStateSuite) TestGetAllMachineBasesNotFound(c *tc.C) {
```

Comment:
Suggest a rename of this test. NotFound can imply to me at least that we expect a not found condition to arise. Something like `TestGetAllMachineBasesNoMachinesInModel`